src/ImageData/CartaFitsImage.cc

```diff
@@ -318,7 +318,7 @@ void CartaFitsImage::SetUpImage() {
         setCoordinateInfo(coord_sys);
 
         // Set image units
-        setUnits(casacore::ImageFITSConverter::getBrightnessUnit(unused_headers, log));
+        setUnits(GetBrightnessUnit(unused_headers));
```

Comment:
Instead of copying this entire function from casacore, would it not be possible to preprocess `unused_headers` to replace this value in-place before calling the casacore function?